agent-common/src/com/thoughtworks/go/agent/launcher/ServerBinaryDownloader.java

```diff
@@ -17,161 +17,134 @@
+                LOG.error("Couldn't update " + downloadableFile + ". Sleeping for 1m. Error: ", e);
                 try {
-                    headers = headers();
-                    File localFile = new File(downloadableFile.getLocalFileName());
-                    md5 = headers.get(MD5_HEADER);
-                    sslPort = headers.get(SSL_PORT_HEADER);
-                    if (!localFile.exists() || !checksOut(localFile, md5)) {
-                        PerfTimer timer = PerfTimer.start("Downloading new " + downloadableFile + " with md5 signature: " + md5);
-                        downloaded = download();
-                        timer.stop();
-                    }
-                    updated = true;
-                } catch (Exception e) {
-                    e.printStackTrace();
-                    LOG.error("Couldn't update " + downloadableFile + ". Sleeping for 1m. Error: " + e.toString());
-                    try {
-                        int period = Integer.parseInt(System.getProperty("sleep.for.download", "60000"));
-                        Thread.sleep(period);
-                    } catch (InterruptedException ie) { /* we don't care. Stupid checked exception.*/ }
-                }
+                    int period = Integer.parseInt(System.getProperty("sleep.for.download", "60000"));
+                    Thread.sleep(period);
+                } catch (InterruptedException ie) { /* we don't care. Stupid checked exception.*/ }
             }
-            return downloaded;
         }
+        return downloaded;
     }
 
-    Map<String, String> headers() throws Exception {
-        Map<String, String> headers = serverCall.invoke(new HttpHead(checkUrl()), rootCertFile, sslVerificationMode).headers;
-        if (!headers.containsKey(MD5_HEADER)) {
-            LOG.error(format("Contacted server at URL %s but the server did not send back a response containing the header %s", downloadableFile.url(urlGenerator), MD5_HEADER));
+    void fetchUpdateCheckHeaders(DownloadableFile downloadableFile) throws Exception {
+        String url = downloadableFile.validatedUrl(urlGenerator);
+        final HttpRequestBase request = new HttpHead(url);
+        request.setConfig(RequestConfig.custom().setConnectTimeout(HTTP_TIMEOUT_IN_MILLISECONDS).build());
+
+        try (
+                CloseableHttpClient httpClient = httpClientBuilder.build();
+                CloseableHttpResponse response = httpClient.execute(request)
+        ) {
+            handleInvalidResponse(response, url);
+            this.md5 = response.getFirstHeader(MD5_HEADER).getValue();
+            this.sslPort = response.getFirstHeader(SSL_PORT_HEADER).getValue();
         }
-        return headers;
     }
 
-    private String checkUrl() {
+    protected boolean download(final DownloadableFile downloadableFile) throws Exception {
+        File toDownload = downloadableFile.getLocalFile();
+        LOG.info("download of " + toDownload + " started at " + new Date());
```

Comment:
Capitalize this properly? Also, if we're using a logger, can we rely on the timestamp in the logger, instead of printing the Date?